### DISCUSSION TITLE
Postgres: Improve error handling in configuration

### DISCUSSION THREAD

**Discussion by pallavi-shirude2022** [grafana/grafana#48975] [Data Sources]
Please use this template to create your bug report. By providing as much info as possible you help us understand the issue, reproduce it and resolve it for you quicker. Therefore take a couple of extra minutes to make sure you have provided all info needed.

PROTIP: record your screen and attach it as a gif to showcase the issue.

- Questions should be posted to: https://community.grafana.com
- Use query inspector to troubleshoot issues: https://bit.ly/2XNF6YS
- How to record and attach gif: https://bit.ly/2Mi8T6K
-->

**What happened**:
When I
 try connecting to Aurora RDS instance using Postgresplugin, it shows on console that the DATABSE CONNECTION OK.
But actually in backend its failing with 400 error .
Please see attached HAR file 

**What you expected to happen**:
Plugin shall not give "DATABASE CONNECTION ok " message when its actually failing in backend.

**How to reproduce it (as minimally and precisely as possible)**:
Just do below:
1) spin up one RDS aurora instance after its active, go to Grafana (I used AWS managed Grafana), and 
2)  Add datasource
3)select postgres plugin
4) give endpoint:port in host, database name , user and password, Kept SSL disabled .
5) kept other settings as is 
5) Click on "Save and Test" (try to select database that is not reachable and shall actually fail, for my case i selected RDS Aurora myslql instance for postgres plugin with endpoint:3306)
6) the message on console after "save and test" is : "Database Connection OK"
7) if you see HAR file in networking response was actually 400 error.

**Anything else we need to know?**:

**Environment**:
- Grafana version: AWS managed Grafana
- Data source type & version: RDS aurora -mysql
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins: postgres
- Others:
![g-833400b1a6 grafana-workspace eu-west-1 amazonaws com_Archive  22-05-07 10-49-39  - Copy har](https://user-images.githubusercontent.com/105088908/167232282-99974057-58b0-43d9-8130-f64f4027e1e3.gif)

https://user-images.githubusercontent.com/105088908/167232323-583bce4b-3893-4964-8318-a093fcd18af4.mp4

**Comment by zoltanbedi** [comment]
@pallavi-shirude2022 thanks for letting us know. This is fixed in 8.3.4 https://github.com/grafana/grafana/pull/43762

**Comment by gabor** [comment]
as this has been fixed, closing it.
